security/cosign/Portfile

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           golang 1.0
+
+go.setup            github.com/sigstore/cosign 0.5.0 v
+revision            0
+
+categories          security
+maintainers         {outlook.de:judaew @judaew} openmaintainer
+license             Apache-2
+
+description         Container Signing
+
+long_description    \
+    Container Signing, Verification and Storage in an OCI registry. Cosign \
+    aims to make signatures invisible infrastructure.
+
+checksums           rmd160  2c293692ecc24204b76042b60bb3a1850247b299 \
+                    sha256  126b4a6081ddab6cb0b42760e2f25126d69e17064dbaa1b0b7cf2a045439ed9d \
+                    size    6005926
+
+# FIXME: This port currently can't be built without allowing go mod to fetch
+build.env-delete    GOPROXY=off GO111MODULE=off
+
+build.pre_args-append \
+    -ldflags '-X github.com/sigstore/cosign/cmd/cosign/cli.gitVersion=${version}'
+build.args          -o build/ ./...
+
+destroot {
+    xinstall -m 0755 ${worksrcpath}/build/cosign ${destroot}${prefix}/bin/
+
+    xinstall -d ${destroot}${prefix}/share/doc/${name}
+    xinstall -m 0644 -W ${worksrcpath} \
+        CHANGELOG.md COPYRIGHT.txt EXAMPLES.md FUN.md KEYLESS.md KMS.md \
+        LICENSE RELEASES.md SPEC.md TOKENS.md USAGE.md \
+        ${destroot}${prefix}/share/doc/${name}
+}
+
+variant sget description {sget command for safer, automatic verification of signatures and integration with transparency log Rekor} {
+    post-destroot {
+        xinstall -m 0755 ${worksrcpath}/build/sget \
+            ${destroot}${prefix}/bin/
+    }
+}
+
+variant copasetic \
+    description {experimental OPA plugin (embedded OPA interpreter)} {
+    post-destroot {
+        xinstall -m 0755 ${worksrcpath}/build/copasetic \
+            ${destroot}${prefix}/bin/
+    }
+}
```

Comment:
why the use of variants here? It seem you don't need any additional dependencies for this and the files are built anyways; if so, why not just install them unconditionally?

Reply:
The binaries are about ~50 MB each in size. `cosign` alone covers almost the entire functionality. The rest of the binaries are needed for specific cases. For example, ArchLinux only installs `cosign` and `copasetic` (experimental), while NixOS only installs `cosign`. With the help of variants, I want to leave the opportunity to install binaries in case the user needs them.

Reply:
okay, I didn't realize the binaries were quite big - that makes sense.